### PR TITLE
Implementation of Full Screen mode for SwingUI and the possibility to switch between Full Screen and Windowed.

### LINKED ISSUE #399: Make it possible to switch to full screen mode
ALT + ENTER should be the shortcut to do this.

### PR DESCRIPTION
**Pull request by andreas-eberle**
This resolves #399.

### CODE REVIEW

**Comment by andreas-eberle** [comment]
@michaelzangl @codingberlin Can you have a look at the changes?

**Comment by andreas-eberle** [comment]
Should we make the full screen mode the default behavior?

**Comment by codingberlin** [comment]
My next days are quite full. Please give me some time to check the full screen on Mac.

File: jsettlers.main.swing/src/main/java/jsettlers/main/swing/JSettlersFrame.java [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
  * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
  * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
  * DEALINGS IN THE SOFTWARE.
  *******************************************************************************/
 package jsettlers.main.swing;
 
-import java.awt.Container;
-import java.awt.Dimension;
-import java.awt.HeadlessException;
+import java.awt.*;
```
Comment (michaelzangl) [inline jsettlers.main.swing/src/main/java/jsettlers/main/swing/JSettlersFrame.java]: Please use the eclipse formatter.

File: jsettlers.graphics/src/main/java/jsettlers/graphics/startscreen/SettingsManager.java [hunk 2]
```diff
@@ -1,12 +1,12 @@
 /*******************************************************************************
  * Copyright (c) 2015
- *
+ * <p/>
```
Comment (michaelzangl) [inline jsettlers.graphics/src/main/java/jsettlers/graphics/startscreen/SettingsManager.java]: Remove these.

File: jsettlers.main.swing/src/main/java/jsettlers/main/swing/JSettlersFrame.java [hunk 3]
```diff
@@ -55,21 +55,62 @@
...
+						switchFullScreenMode();
+						return true; // consume this key event.
+					}
+				}
+				return false;
+			}
+		});
+	}
+
+	private void switchFullScreenMode() {
```
Comment (michaelzangl) [inline jsettlers.main.swing/src/main/java/jsettlers/main/swing/JSettlersFrame.java]: switchFullScreenMode -> toggleFullScreenMode

**Comment by andreas-eberle** [comment]
@codingberlin: That's ok

**Comment by andreas-eberle** [comment]
@michaelzangl: Can you show the short commit hash somewhere when the game is running? E.g in the lower right corner? This way we always see the used commit in screenshots, even if the user plays in full screen mode.

@codingberlin: For the main swing menu we should probably add something simmilar.

**Comment by codingberlin** [comment]
Wow the fullscreen is quite impressive. It fells very awesome.
(So it worked on Mac)

I would suggest to add something (maybe a checkbox) to the SettingsDialog to toggle the fullscreen mode.

Currently I'm working at another branch for the AI to address some issues ... and you might guess it ... to make it even stronger :-)
When I'm finished with that I could support here. But feel free to finish this issue earlier if you like.

**Comment by michaelzangl** [comment]
I suggest using `F11` as keybinding.

**Comment by andreas-eberle** [comment]
@codingberlin: Thanks for testing. I also think an option in the settings menu would be great. 

@michaelzangl: Why F11? In every modern game, ALT+ENTER is the full screen keybinding (at least on Windows).

**Comment by andreas-eberle** [comment]
@andreasb242: Can you create us a skinned JCheckBox that we could add to the settings dialog?

**Comment by andreas-eberle** [comment]
I think we can make full screen the default behavior for new players. If someone wants to disable it, he can with the keybinding and later with a checkbox in the settings. What do you think?

**Comment by michaelzangl** [comment]
I would like to wait with this until we have the settings.
The main issue is that we do not have any visual quit button in our game - so people that do not know the key bindings might have problems with this.

**Comment by andreas-eberle** [comment]
We actually have the original quit button (the computer icon in the lower left corner).

**Comment by michaelzangl** [comment]
But we do not have any way to quit full screen mode.

**Comment by andreasb242** [comment]
Ok, I'll create an UI for the Checkbox. In the original game is no Checkbox in the menu, right?

>The main issue is that we do not have any visual quit button in our game - so people that do not know
>the key bindings might have problems with this.

What about display all the key shortcuts if you press F1?

e.g. Ubuntu has a such a Dialog, if you do a fresh install, looking something like this (googled screenshot):
http://i.stack.imgur.com/dX0jc.png

**Comment by michaelzangl** [comment]
Well, then you need some way to know about the F1 key ;-)

**Comment by andreasb242** [comment]
Add a button in the "Quit" menu, there should be enough space for an additional button?

**Comment by andreas-eberle** [comment]
@andreasb242: The idea to somehow display the shortcuts is a good idea. But maybe we simply add a menu to configure the shortcuts... but for me that's something to be done in the future.

Regarding quitting the game: You don't need the shortcut. We have a quit button. See screenshot

unbenannt

As this is like it was in the original game, I don't see the problem that a user could not quit the game...

**Comment by michaelzangl** [comment]
In my opinion, this is ready to be merged. The visible settings can be done later.

**Pull request merged**